### PR TITLE
Added kron for Diagonal matrices

### PR DESCRIPTION
**Pull request by jebej**

### CODE REVIEW

**Review by Sacha0 — APPROVED** [review]
LGTM from a rapid skim! Thanks @jebej! :)

File: stdlib/LinearAlgebra/src/diagonal.jl [hunk 1]
```diff
@@ -354,6 +354,16 @@ rdiv!(A::AbstractMatrix{T}, transD::Transpose{<:Any,<:Diagonal{T}}) where {T} =
 \(adjF::Adjoint{<:Any,<:Factorization}, D::Diagonal) =
     (F = adjF.parent; ldiv!(adjoint(F), Matrix{typeof(oneunit(eltype(D))/oneunit(eltype(F)))}(D)))
 
+function kron(A::Diagonal{T1}, B::Diagonal{T2}) where {T1<:Number, T2<:Number}
+    valA = A.diag; nA = length(valA)
+    valB = B.diag; nB = length(valB)
+    valC = Vector{typeof(zero(T1)*zero(T2))}(nA*nB)
```
Comment (jebej) [inline stdlib/LinearAlgebra/src/diagonal.jl]: add `undef` here

**Comment by jebej** [comment]
This is ready to be merged, although it could wait after 0.7 since it's not a breaking change (unless someone was relying on the fact that this used to return an `Array` and not a `Diagonal`).

**Comment by StefanKarpinski** [comment]
No harm merging once it passes tests by the same token—it doesn't break anything and it's uncontroversial, so why not?

**Pull request merged**